contracts/vaults/Claim.sol

```diff
@@ -26,41 +27,34 @@ contract Claim is Base {
     * @notice Called by a committee to submit a claim for a bounty.
     * The submitted claim needs to be approved or dismissed by the Hats governance.
     * This function should be called only on a safety period, where withdrawals are disabled.
-    * Upon a call to this function by the committee the pool withdrawals will be disabled
+    * Upon a call to this function by the committee the vault's withdrawals will be disabled
     * until the Hats governance will approve or dismiss this claim.
-    * @param _pid The pool id
     * @param _beneficiary The submitted claim's beneficiary
     * @param _bountyPercentage The submitted claim's bug requested reward percentage
     */
-    function submitClaim(uint256 _pid,
-        address _beneficiary,
-        uint256 _bountyPercentage,
-        string calldata _descriptionHash)
+    function submitClaim(address _beneficiary, uint256 _bountyPercentage, string calldata _descriptionHash)
     external
-    onlyCommittee(_pid)
-    noActiveClaims(_pid)
+    onlyCommittee()
+    noActiveClaim()
     {
         if (_beneficiary == address(0)) revert BeneficiaryIsZero();
+        HATVaultsRegistry.GeneralParameters memory generalParameters = registry.getGeneralParameters();
         // require we are in safetyPeriod
         // solhint-disable-next-line not-rely-on-time
         if (block.timestamp % (generalParameters.withdrawPeriod + generalParameters.safetyPeriod) <
         generalParameters.withdrawPeriod) revert NotSafetyPeriod();
-        if (_bountyPercentage > bountyInfos[_pid].maxBounty)
+        if (_bountyPercentage > maxBounty)
             revert BountyPercentageHigherThanMaxBounty();
-        uint256 claimId = uint256(keccak256(abi.encodePacked(_pid, block.number, nonce++)));
-        claims[claimId] = Claim({
-            pid: _pid,
+        activeClaim = Claim({
```

Comment:
we added a claimId to be able to track the claim in the arbitrator. So please restore that logic.

Reply:
But there is only one claim per vault, so you can just as easily use the vault address... it just makes no sense to pass claim ID everywhere if there's only one claim in the contract, it's just extra data with no logic or benefit.

Reply:
it is because the arbitrator may be async. It is written down somewhere selse, but the scenario roughly is:
- active claim gets disputed (call this claim 1)
- arbittrator takes a long time to decide
- active claim gets resolved bc of arbitrator activity on the vaults contract
- a new claim is made (call this claim 2)
- arbitrator finally decides on result of claim 1
- arbitrator now approves "the active claim", but this is claim 2 and that is Really Wrong

Reply:
But if it's a different claim it will not be challenged, so a delayed call will just fail anyway

Reply:
well the only reason it is not challenged in my scenario si bc I forgot to add that step :) - I edit my response

Reply:
But the arbitrator would have to challenge claim 1, then claim 2 in your scenario, which practically acknowledges that there is a new claim.

Reply:
also, independent of this unlikely (but afaics possible :))  scenario, these claims are objects being passed around in different systems (arbitrator, ui systems) and it will be useful to give them identifiers. Just think of how shitty it is to develop the frontend and yuo have to check the status of a certain claim without an id...

Reply:
If the arbitrator challenges, then challenges again the obviously they know there is a new claim

Reply:
But there is only ever one claim for each vault address. I mean I could add back this claim ID but it still just seems to me like a waste of gas with no added benefit...

Reply:
> If the arbitrator challenges, then challenges again the obviously they know there is a new claim

This issue came up in the kleros prototype integration here: https://github.com/hats-finance/hats-contracts/pull/133 and so it is not so obvious at all. Kleros will need an ID anyway (I guess any arbitration mechanism would want that) so the question there was more of where it should be generated, and at the vaults seems the most straightforward.

> But there is only ever one claim for each vault address. I 

There is only ever one claim for each vault address **in the vault contract**. But for the arbitrator and for the UI, there will be the entire history of claims. And they will need a way to check which of those claims is "the active claim".

It is possible to resolve these issues in the abitrator integration and in the ui (by creating ids there and keeping careful track of the status in the vaults to know which one of the claims is "the active claim"). But it will be a big pain :)

Reply:
Ok I added the claim Id: https://github.com/hats-finance/hats-contracts/pull/276/commits/aa1ed0cff73feb374b99d385e126fb01b8b0617e

But it significantly increases gas cost and contract size, while it could be implemented in the UI, so I'm still against having that :)